### PR TITLE
Define User#undestroy as alias for restore with options.

### PR DESCRIPTION
**Pull request by ashercodeorg**

### CODE REVIEW

File: dashboard/test/models/user_test.rb [hunk 1]
```diff
@@ -1969,6 +1970,22 @@ def mock_geocoder_result(result)
     assert student.reload.deleted?
   end
 
+  test 'undestroy restores recent dependents only' do
+    teacher = create :teacher
+    old_section = create :section, teacher: teacher
+    Timecop.freeze 1.hour.ago do
+      old_section.destroy!
```
Comment (ashercodeorg) [inline dashboard/test/models/user_test.rb:1977]: Not the cleanest that the section is being destroyed before it was created. But such is the nature of time travel.

Comment (aoby) [reply dashboard/test/models/user_test.rb:1977]: image

**Comment by ashercodeorg** [comment]
Actually, PTAL, I am now throwing an exception if the user is purged.

File: dashboard/app/models/user.rb [hunk 2]
```diff
@@ -1586,6 +1586,14 @@ def associate_with_potential_pd_enrollments
     end
   end
 
+  # Via the paranoia gem, undelete / undestroy the deleted / destroyed user and any (dependent)
+  # destroys done around the time of the delete / destroy.
+  def undestroy
+    raise 'Unable to restore a purged user' if purged_at
+    # Paranoia documentation at https://github.com/rubysherpas/paranoia#usage.
```
Comment (aoby) [inline dashboard/app/models/user.rb]: nit: newline before the comment

Comment (ashercodeorg) [reply dashboard/app/models/user.rb]: Done.

**Pull request merged**